cmd/liquidator.go

```diff
@@ -0,0 +1,138 @@
+package cmd
+
+import (
+	"context"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// LiquidationTarget contains the address of a borrower who is over their borrow limit, as
+// well as their current borrowed amounts and collateral in all denominations.
+// CONTRACT: Collateral uTokens must be expressed as their equivalent value in base tokens.
+type LiquidationTarget struct {
+	Addr       sdk.AccAddress
+	Borrowed   sdk.Coins
+	Collateral sdk.Coins
+}
+
+// LiquidationOrder expresses the intent to perform, or the outcome of, a liquidation.
+// Collateral reward amounts are always expressed in base tokens equivalent to their
+// underlying uToken amount.
+type LiquidationOrder struct {
+	Addr   sdk.AccAddress
+	Repay  sdk.Coin
+	Reward sdk.Coin
+}
+
+// startLiquidator causes the liquidator to continuously look for liquidation targets, decide on which
+// borrowed and collateral denominations to attempt to liquidate, and attempt to execute any
+// liquidations whose estimated outcomes are approved by its configured decisionmaking. Returns
+// only when context is cancelled.
+func startLiquidator(
+	ctx context.Context,
+	cancelFunc context.CancelFunc,
+) error {
+
+	// loop as long as ctx is not cancelled
+	for ctx.Err() == nil {
```

Comment:
This doesn't seem idiomatic to me. Can you consider instead the following:

```go
ticker := time.NewTicker(time.Minute)
defer ticker.Stop()

for {
  select {
    case <- ctx.Done():
        if err := ctx.Err(); err != nil {
            // handle error
        }

    case <-ticker.C:
        // perform core logic
  }
}
```

Reply:
done at [b1afef4](https://github.com/umee-network/liquidator/pull/1/commits/b1afef47e55513f5a26513593e6b06a98492153f)